src/test/java/org/jitsi/meet/test/base/ParticipantOptions.java

```diff
@@ -55,12 +34,18 @@ static ParticipantType getParticipantType(
     /**
      *
      */
-    public void load(Properties config, String configPrefix)
+    public void load(
+        Properties config,
+        String configPrefix,
+        ParticipantOptions overrides)
     {
-        this.configPrefix = configPrefix;
+        this.setProperty(TYPE_PROP, getParticipantType(config, configPrefix));
 
         String name = configPrefix.substring(configPrefix.indexOf('.') + 1);
         this.setProperty(NAME_PROP, name);
+
+        if (overrides != null)
```

Comment:
Can we add "override" or "merge" method and move the override there ? Those overrides normally should be of no interest to the load method (to not make things more complex and encourage to write some weird stuff here). I see few other things that can be simplified, but I don't want to torture you with that anymore. I'll do that in a separate PR eventually. If the tests will pass it should be good enough for now. But please remember about missing JavaDocs...

Reply:
Check it now, hope all javadocs are there. The tests pass (just the LockRoom but it is fixed in master).